### PR TITLE
Big datagrams (>64K) support in RTPS layer <master> [7372]

### PR DESCRIPTION
**Pull request by adolfomarver**
Current RTPS implementation do not support transports capable of sending datagrams bigger than 64K. This requirement implements the necessary changes to the library.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1236/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1022/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1233/)

File: src/cpp/rtps/messages/submessages/DataMsg.hpp [hunk 1]
```diff
@@ -35,20 +35,22 @@ bool RTPSMessageCreator::addMessageData(
...
-        const EntityId_t& readerId,
-        bool expectsInlineQos,
-        InlineQosWriter* inlineQos)
+    CDRMessage_t* msg,
+    const CacheChange_t* change,
+    TopicKind_t topicKind,
+    const EntityId_t& readerId,
+    bool expectsInlineQos,
+    InlineQosWriter* inlineQos,
+    bool* is_big_submessage)
```
Comment (MiguelCompany) [inline src/cpp/rtps/messages/submessages/DataMsg.hpp]: Wrong indentation (should be 8 spaces)

File: src/cpp/rtps/writer/RTPSWriter.cpp [hunk 2]
```diff
@@ -148,6 +148,16 @@ CONSTEXPR uint32_t info_dst_message_length = 16;
...
+uint32_t RTPSWriter::get_max_fragment_payload_size()
+{
+    constexpr uint32_t data_frag_header_size = 28;
+    constexpr uint32_t max_inline_qos_size = 32;
+    // Max fragment is 64KBytes_max - header - inlineqos - 4(for better alignment)
+    constexpr uint32_t max_fragment_size = std::numeric_limits<uint16_t>::max() - data_frag_header_size - max_inline_qos_size - 4;
+    
+    return max_fragment_size;
+}
+
```
Comment (MiguelCompany) [inline src/cpp/rtps/writer/RTPSWriter.cpp]: Move this method to the header and make it an inline constexpr one.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1248/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1035/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1245/)

File: src/cpp/rtps/messages/RTPSMessageGroup.cpp [hunk 3]
```diff
@@ -37,73 +37,78 @@ namespace rtps {
...
     return(c1->sequenceNumber < c2->sequenceNumber);
 }
 
-bool sort_SeqNum(const SequenceNumber_t& s1,const SequenceNumber_t& s2)
+bool sort_SeqNum(
+        const SequenceNumber_t& s1,
+        const SequenceNumber_t& s2)
 {
     return(s1 < s2);
 }
```
Comment (MiguelCompany) [inline src/cpp/rtps/messages/RTPSMessageGroup.cpp]: Unused local file functions. Remove them.

File: src/cpp/rtps/messages/RTPSMessageGroup.cpp [hunk 4]
```diff
@@ -139,16 +146,19 @@ void prepare_SequenceNumberSet(std::set<SequenceNumber_t>& changesSeqNum,
     }
 }
 
-bool compare_remote_participants(const std::vector<GUID_t>& remote_participants1,
+bool compare_remote_participants(
```
Comment (MiguelCompany) [inline src/cpp/rtps/messages/RTPSMessageGroup.cpp]: Unused local file function. Remove it.

File: src/cpp/rtps/messages/RTPSMessageGroup.cpp [hunk 5]
```diff
@@ -157,33 +167,47 @@ bool compare_remote_participants(const std::vector<GUID_t>& remote_participants1
...
     participants.clear();
 
     for (const GUID_t& endpoint : endpoints)
     {
         if (std::find(participants.begin(), participants.end(), endpoint.guidPrefix) == participants.end())
+        {
             participants.push_back(endpoint.guidPrefix);
+        }
     }
 }
```
Comment (MiguelCompany) [inline src/cpp/rtps/messages/RTPSMessageGroup.cpp]: Unused local file functions. Remove them.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1261/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1048/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1258/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1327/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1105/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1324/)

File: include/fastdds/rtps/messages/RTPSMessageGroup.h [hunk 6]
```diff
@@ -44,195 +44,212 @@ class RTPSMessageGroup_t;
...
+
+    uint32_t currentBytesSent_;
+
+    GuidPrefix_t current_dst_;
 
 #if HAVE_SECURITY
-        CDRMessage_t* encrypt_msg_;
+    RTPSParticipantImpl* participant_;
+
+    CDRMessage_t* encrypt_msg_;
```
Comment (MiguelCompany) [inline include/fastdds/rtps/messages/RTPSMessageGroup.h]: `participant_` is also used without security. This does not build with security off.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1515/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1287/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1513/)

**Pull request merged**